UKHomeOffice/nmsw-ui
# NMSW-201 Sign in with credentials

# Ticket

NMSW-201

----

## AC

GIVEN I have entered a correct email address and password
WHEN I sign in
THEN I am taken to the dashboard
 
GIVEN I have NOT entered a correct email address and password
WHEN I sign in
THEN a generic error message is displayed (Email and password combination is invalid, no fields highlighted)
 
GIVEN I have entered an email address for an unverified email address
WHEN I sign in
THEN I am taken to the request new verification link page (Confirmed with UCD team)
 
GIVEN the email address has an invalid format
WHEN I submit the form
THEN an error message is displayed next to the field (Enter a valid email address, email field highlighted)

----

## To test

- Run app
- Got to localhost:3000
- Create an account
- Verify your account and complete the account creation process
- Go to /sign-in
- Click submit
- > You should see a fields are required error
- Enter an invalid format email
- Click submit
- > You should see an invalid email error
- Enter a valid email address
- Enter an incorrect password
- Click submit
- > You should see a combination is invalid error
- Type into either the email or password field
- The API error message should disappear
- Create another account
- Do NOT click on the email link to verify it
- Go to /sign-in
- Enter the email you have NOT verified
- > You should be taken to the /create-account/request-new-verification-link page with the email address you entered filled in
- Go back to /sign-in
- Enter a valid email and password
- Click submit
- > You should be taken to the dashboard
- > You should see the sign out link in the nav

----

## Notes
